docs/rules/no-whitespace-before-property.md

```diff
@@ -0,0 +1,53 @@
+# Disallow whitespace before properties (no-whitespace-before-property)
+
+Disallows whitespace around dot or brackets before properties of objects if they are on the same line.
+
+## Rule Details
+
```

Comment:
Need a sentence here in the form "This rule aims to (what does it do) and as such will report a problem when (when does it report a problem)."